aws/data_source_aws_sfn_state_machine.go

```diff
@@ -0,0 +1,98 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/sfn"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsSfnStateMachine() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsSfnStateMachineRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"arn": {
```

Comment:
This attribute is missing documentation and is not implemented below (it should be though 👍 as its generally expected by Terraform AWS Provider users, even with the `id` attribute).

Reply:
added